tests/expr/name/map_test.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from contextlib import nullcontext as does_not_raise
+from typing import Any
+
+import polars as pl
+import pytest
+
+import narwhals as nw
+from tests.utils import compare_dicts
+
+data = {"foo": [1, 2, 3], "BAR": [4, 5, 6]}
+
+
+def map_func(s: str | None) -> str:
+    return str(s)[::-1].lower()
+
+
+def test_map(constructor_with_lazy: Any) -> None:
+    df = nw.from_native(constructor_with_lazy(data))
+    result = df.select((nw.col("foo", "BAR") * 2).name.map(function=map_func))
+    expected = {map_func(k): [e * 2 for e in v] for k, v in data.items()}
+    compare_dicts(result, expected)
+
+
+def test_map_after_alias(constructor_with_lazy: Any) -> None:
+    df = nw.from_native(constructor_with_lazy(data))
+    result = df.select((nw.col("foo")).alias("alias_for_foo").name.map(function=map_func))
+    expected = {map_func("foo"): data["foo"]}
+    compare_dicts(result, expected)
```

Comment:
These tests `<>_after_alias` are to make sure to account for what's been discussed in #186 [comment](https://github.com/narwhals-dev/narwhals/pull/186#issuecomment-2123146302)